firmware/Core/Src/rtos_tasks/rtos_tasks.c

```diff
@@ -166,3 +167,52 @@ void TASK_execute_telecommands(void *argument) {
 
 	} /* End Task's Main Loop */
 }
+
+void TASK_monitor_freertos_highstack_watermarks(void *argument) {
+	TASK_HELP_start_of_task();
+
+	while (1) {
+
+		for(uint32_t x = 0; x < task_handles_array_size; x++){
+
+			if(task_handles_array[x].task_handle != NULL)
+			{
+				// Dereferencing the task_handle pointer
+				osThreadId_t dereferenced_task_handle = *(task_handles_array[x].task_handle);
+
+				// Determine the threshold of the task
+				uint32_t task_threshold_words = (task_handles_array[x].stack_size_bytes * CONFIG_highstack_watermark_percentage_threshold) / 400;
+
+				// Get the highstack watermark
```

Comment:
Is this going to constanty spew warnings after it passes the threshold? 

Might be fine, but might be nice if there was any easy way to only warn if it gets even worse (or something like that).

Reply:
Thats true.

So we need to keep track of when and the remaining stack based on the `task_highstack_watermark_bytes` and do a comparison between `task_highstack_watermark_bytes` and let us say `last_reported_task_highstack_watermark_bytes`

If the `task_highstack_watermark_bytes` is smaller than `last_reported_task_highstack_watermark_bytes` then we print to UART the warning?

Reply:
![image](https://github.com/user-attachments/assets/c5e7b5d2-20c5-4e78-93c7-a3cfef6710d0)

For the `TASK_DEBUG_print_heartbeat` its worst remaining stack size is always 0 so, it will log a warning only once if I set the condition to be less than `last_reported_task_highstack_watermark_bytes` and always if I set it to be less than or equal to `last_reported_task_highstack_watermark_bytes`

Reply:
If you have an idea of how to go about it, I am open to suggestions. Otherwise, for now, I could move forward with the idea I put above.

Reply:
Done